File: sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum=0;
    void traverse(TreeNode* root,string temp){
        if(root==NULL) return;
        if(root->left) traverse(root->left,temp+to_string(root->val));
        if(root->right) traverse(root->right,temp+to_string(root->val));
        if(!root->left and !root->right){
            sum+=stoi(temp+to_string(root->val));
            return;
        }
    }
    int sumNumbers(TreeNode* root) {
        traverse(root,"");
        return sum;
    }
};

File: sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0;
    public int sumNumbers(TreeNode root) {
        if(root==null) return 0;
        traverse(root,"");
        return sum;
    }
    public void traverse(TreeNode root,String currentSum){
        if(root==null) return;
        if(root.left!=null) traverse(root.left,currentSum+Integer.toString(root.val));
        if(root.right!=null) traverse(root.right,currentSum+Integer.toString(root.val));
        if(root.right==null && root.left==null){
            sum+=Integer.parseInt(currentSum+root.val);
            return;
        }
    }
}